validate/validate.py

```diff
@@ -10,40 +10,31 @@
     path_readable,
     path_writable
 )
-from generate_checksum.checksum import validate_checksums
 
 # Currently supported data types
 DIR_TYPES = ['directory-r', 'directory-rw']
 FILE_TYPES_DICT = {'file-bam': ['.bam', '.cram', '.sam'], 'file-vcf': ['.vcf', '.vcf.gz'],
     'file-fasta': ['.fasta', '.fa'], 'file-fastq':['.fastq', '.fq.gz', '.fq', '.fastq.gz'],
     'file-bed': ['.bed', '.bed.gz'], 'file-py': ['.py']}
-GENERIC_FILE_TYPE = 'file-input' # file type needs to be detected
 UNKNOWN_FILE_TYPE = 'file-unknown' # file type is unlisted
 CHECK_FUNCTION_SWITCH = {
     'file-bam': check_bam,
     'file-vcf': check_vcf
 }
-CHECK_COMPRESSION_TYPES = ['file-vcf', 'file-fastq', 'file-bed']
+CHECK_COMPRESSED = ['file-vcf', 'file-fastq', 'file-bed']
+COMPRESSION_TYPES = ['.gz']
 
-def validate_file(path:Path, file_type:str):
+def validate_file(path:Path, detected_file_type:str, file_extension:str):
     ''' Validate a single file '''
     path_exists(path)
 
-    detected_file_type, file_extension = detect_file_type_and_extension(path)
     if not file_extension:
         raise TypeError(f'File {path} does not have a valid extension.')
-    if (detected_file_type != UNKNOWN_FILE_TYPE and
-        file_type != GENERIC_FILE_TYPE and
-        detected_file_type != file_type):
-        raise ValueError(f'Indicated and detected file types do not match. '\
-            f'Indicated: {file_type}, detected: {detected_file_type}'
-        )
-
-    if detected_file_type in CHECK_COMPRESSION_TYPES:
+    if detected_file_type == UNKNOWN_FILE_TYPE:
+        raise TypeError(f'File {file_extension} is not supported')
+    if detected_file_type in CHECK_COMPRESSED:
         check_compressed(path, file_extension)
 
-    validate_checksums(path)
```

Comment:
We'll still want to keep this; it should be working. This is basically a generic checksum check if they exist

Is it broken?

Reply:
When I built the image it couldn't find the module. But it didn't appear incorrect to me in the import statement, so I don't know what happened.

Reply:
Ahh the issue is that I didn't update the Dockerfile in my previous PR; the Dockerfile only copies the `validate` directory currently so the entire checksum module isn't copied to the image

Reply:
that makes sense. I'll put the deleted checksum code back then. we can fix the docker file in another PR and I can just merge the changes into this one if I'm not done yet.

Reply:
That's fine with me!